libs/core-ui/src/lib/Context/ModelAssessmentContext.tsx

```diff
@@ -112,6 +112,10 @@ export interface IModelAssessmentContext {
   requestExp?:
     | ((index: number, abortSignal: AbortSignal) => Promise<any[]>)
     | undefined;
+  requestForecast?: (
+    request: any[],
+    abortSignal: AbortSignal
+  ) => Promise<number[]>;
```

Comment:
Isn't this same as request prediction?

Reply:
Not quite. We have to be able to provide transformations on a time series. In other words, we provide the transformation operation (let's say method = "Add", value = 5, feature = "abc") so this needs to be applied to all data points that are part of this time series and then we need to get the predictions. Doing this in the backend seems smarter to me than duplicating the data in the UI and applying the transformation before sending all that data across. 

The "other" what-if (for tabular data) creates individual transformations that don't generalize beyond a single point so perhaps there isn't much of a difference there and that's why it uses `requestPredictions`.
Wdyt?

Reply:
Settled on this offline. Marking as resolved.